collator/src/lib.rs

```diff
@@ -461,51 +512,52 @@ fn compute_targets(para_id: ParaId, session_keys: &[ValidatorId], roster: DutyRo
 /// build by the given `BuildParachainContext` and arguments to the underlying polkadot node.
 ///
 /// This function blocks until done.
-pub fn run_collator<P>(
-	build_parachain_context: P,
+pub fn run_collator<P, B>(
```

Comment:
Could this function not use `build_collator` internally?

Reply:
I tried but it doesn't work because build_collator is actually an async fn and I need a service to use `run_service_until_exit`. I could use build_collator and use run_until_exit (which takes a future instead) but then we would like some features like: setting the task_executor in the configuration and also the informant